centos5/Dockerfile

```diff
@@ -14,16 +19,300 @@ RUN localedef -i en_US -f UTF-8 en_US.UTF-8 \
+    && cd .. \
+    && rm -rf libmagic-src libmagic.tar.gz
+
+# Bootstrap which
+RUN /home/linuxbrew/bootstrap/curl/bin/curl -Lo which.tar.gz https://carlowood.github.io/which/which-2.21.tar.gz \
+    && mkdir -p which-src \
+    && tar -xf which.tar.gz -C which-src --strip-components=1 \
+    && cd which-src \
+    && ./configure --prefix=/home/linuxbrew/bootstrap/which \
+    && make \
+    && make install \
+    && cd .. \
+    && rm -rf which-src which.tar.gz
+
+# Bootstrap patch
+RUN /home/linuxbrew/bootstrap/curl/bin/curl -Lo patch.tar.xz https://ftp.gnu.org/gnu/patch/patch-2.7.6.tar.xz \
+    && mkdir -p patch-src \
+    && /home/linuxbrew/bootstrap/tar/bin/tar -xf patch.tar.xz -C patch-src --strip-components=1 \
+    && cd patch-src \
+    && ./configure --prefix=/home/linuxbrew/bootstrap/patch \
+    && make install \
+    && cd .. \
+    && rm -rf patch-src patch.tar.xz
+
+# Bootstrap make
+RUN /home/linuxbrew/bootstrap/curl/bin/curl -Lo make.tar.bz2 https://ftp.gnu.org/gnu/make/make-4.2.1.tar.bz2 \
+    && mkdir -p make-src \
+    && /home/linuxbrew/bootstrap/tar/bin/tar -xf make.tar.bz2 -C make-src --strip-components=1 \
+    && cd make-src \
+    && ./configure --prefix=/home/linuxbrew/bootstrap/make \
+    && make install \
+    && cd .. \
+    && rm -rf make-src make.tar.bz2
+
+# Download and extract portable-ruby for Linux x86_64 from
+# https://homebrew.bintray.com/bottles-portable-ruby
+RUN /home/linuxbrew/bootstrap/curl/bin/curl -Lo portable-ruby.tar.gz \
+     https://homebrew.bintray.com/bottles-portable-ruby/portable-ruby-2.3.3_2.x86_64_linux.bottle.tar.gz \
+     && /home/linuxbrew/bootstrap/tar/bin/tar -xf portable-ruby.tar.gz -C /home/linuxbrew/bootstrap \
+     && ln -s 2.3.3_2 /home/linuxbrew/bootstrap/portable-ruby/current
+
+
+## Stage 2
+##   Install Linuxbrew
+## We're using this stage in order to:
+##   1. make sure we do not depend on system's {make,wget,gcc,perl}
+##   2. slim-down the image
+FROM centos:5
+
+COPY --from=0 /etc/yum.repos.d /etc/yum.repos.d
+
+RUN localedef -i en_US -f UTF-8 en_US.UTF-8 \
+    && useradd -m -s /bin/bash linuxbrew \
+    && echo 'linuxbrew ALL=(ALL) NOPASSWD:ALL' >>/etc/sudoers
+
+USER linuxbrew
+WORKDIR /home/linuxbrew
+COPY --from=0 --chown=linuxbrew:linuxbrew /home/linuxbrew/bootstrap /home/linuxbrew/bootstrap
+
+ENV SHELL=/bin/bash \
+    USER=linuxbrew
+
+ENV PATH=/home/linuxbrew/bootstrap/which/bin:$PATH
+ENV PATH=/home/linuxbrew/bootstrap/patch/bin:$PATH
+ENV PATH=/home/linuxbrew/bootstrap/tar/bin:$PATH
+ENV PATH=/home/linuxbrew/bootstrap/libmagic/bin:$PATH
+ENV PATH=/home/linuxbrew/bootstrap/openssl/bin:$PATH
+ENV PATH=/home/linuxbrew/bootstrap/curl/bin:$PATH
+ENV PATH=/home/linuxbrew/bootstrap/git/bin:$PATH
+ENV PATH=/home/linuxbrew/bootstrap/gettext/bin:$PATH
+ENV PATH=/home/linuxbrew/bootstrap/portable-ruby/current/bin:$PATH
+ENV PATH=/home/linuxbrew/.linuxbrew/bin:/home/linuxbrew/.linuxbrew/sbin:$PATH
```

Comment:
Each `ENV` may create a new Docker layer. I forget whether that behaviour has changed.

Reply:
I'd suggest using the same `--prefix` for all of these packages, so that they all share the same `bin`.